src/libs/Network/SequentialQueue.ts

```diff
@@ -198,6 +198,12 @@ function unpause() {
     const numberOfPersistedRequests = PersistedRequests.getAll().length || 0;
     Log.info(`[SequentialQueue] Unpausing the queue and flushing ${numberOfPersistedRequests} requests`);
     isQueuePaused = false;
+
+    // A READ request will wait until all the WRITE requests are done, using the isReadyPromise promise.
+    // When the queue is paused and then unpaused, we call flush which by defaults recreates the isReadyPromise.
+    // After all the WRITE requests are done, the isReadyPromise is resolved, but since it's a new instance of promise,
+    // the pending READ request never received the resolved callback. That's why we don't want to recreate
+    // the promise when unpausing the queue.
     flush(false);
```

Comment:
> > Also, can we please move the explanatory comment to a doc comment on the function. I think that's a little more helpful, otherwise it's not really clear why the param is needed.

I put the comment here because it explains why we don't want to reset the promise when unpausing the queue, so it's specific to the context of unpausing. I think it'll be a little out of context if I put it on the `flush` function. Or is that fine?

Reply:
It's a matter of preference for sure, but the way I think about it is that more people are going to look at the function declaration of `flush` when trying to figure out what it does, than finding this exact use and the comment. So I think it's more appropriate to add context for what this parameter does at the source of truth for the function, which is its declaration.

Maybe it's bet to have a large comment in the doc comment of the function and then a small one here.

Reply:
> Maybe it's bet to have a large comment in the doc comment of the function and then a small one here.

I thought about this too, but I can't find a good way to explain it yet.

Reply:
Updated

Reply:
Thanks